lib/chef/knife/bmcs_server_show.rb

```diff
@@ -22,6 +38,43 @@ class BmcsServerShow < Knife
              long: '--instance_id LIMIT',
              description: 'The OCID of the server to display. (required)'
 
+      def lookup_compartment_name(compartment_id)
+        compartment = identity_client.get_compartment(compartment_id, {})
+        compartment.data.name
+      end
+
+      def lookup_image_name(image_id)
+        image = compute_client.get_image(image_id, {})
```

Comment:
I believe that image, vcn, and subnet could all be in different compartments and might not be visible to the current user. Definitely an edge case, but ideally we should catch 404s and show empty or a message in that case.

Reply:
I believe this all handled in the latest version of the code I pushed up to my branch.